front/components/assistant/conversation/RetrievalAction.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  Chip,
+  DocumentDuplicateIcon,
+  Spinner,
+  Tooltip,
+} from "@dust-tt/sparkle";
+
+import { classNames } from "@app/lib/utils";
+import {
+  RetrievalActionType,
+  RetrievalDocumentType,
+} from "@app/types/assistant/actions/retrieval";
+
+const PROVIDER_LOGO_PATH: { [provider: string]: string } = {
+  notion: "/static/notion_32x32.png",
+  slack: "/static/slack_32x32.png",
+  google_drive: "/static/google_drive_32x32.png",
+  github: "/static/github_black_32x32.png",
+};
+
+export default function RetrievalAction({
+  retrievalAction,
+  status,
+}: {
+  retrievalAction: RetrievalActionType;
+  status: "loading" | "retrieving" | "success" | "error";
```

Comment:
There is no `error` state for the retrieval action, the agent message will be in charge of displaying the error because errors are tied to the agent message and not action.

Reply:
The state should be infered from the RetrievalActionType

`null` loading
`not null but no documents (params will be set)` retrieving
`not null with params and documents` success 

This should simplify the interface to `RetrievalActionType | null`

I thiiink we can even remove `null` since this initial loading can arguably be managed by the agent message component?

Reply:
Yes the intial loading will have to be handled by the AgentMessage (because it's possible to not have any action, so we'll render it only when we get the query params and will have a generic thinking in the agent message)

Reply:
So the only prop is an retrievalAction and you can infer everything from it :+1:

Reply:
The idea was that the agent message computes the status explicitely (incl. error) and the retrievalAction renders itself according to it

Fine by me to make it implicit via RetrievalActionType, in which case I would allow null documents field in RetrievalActionType, to distinguish from empty array which semantically means "no result" vs "searching" => Ok?

I would leave the initial loading to retrievalaction since it is specific to this action computing its parameters if fine by you? 
the idea behind would be other actions may not have this part, or have a different one with different rendering. Def not strongly held

Reply:
I think our messages crossed here :)
Ok i will out the loading logic

Reply:
Agreed with the necessity of null for documents 👍